build/Dockerfile

```diff
@@ -39,7 +39,8 @@ ARG SET_NUM_PROCESSES_TO_NUM_GPUS=True
 
 USER root
 
-RUN dnf remove -y --disableplugin=subscription-manager \
+RUN dnf update -y \
```

Comment:
The other changes all look good to me but there is often a debate on adding `dnf update` into the Dockerfile. On the one hand, if you DO add it then you address OS level vulnerabilities and don't need to wait for RH to release a new image with the vulns fixed. On the other hand, this means that each build of the image is not deterministic, you don't know exactly what version of deps you will have between builds which can cause side effects. I've seen this go both ways, depends on what we want our security policy to be (how fast we will push updates) and what is expected from us from product teams.

Reply:
Out of curiosity other than the `libexpat` vulnerability, were there others you saw?

Reply:
Good question - it's a point in time scan, so what was scanned 17 days ago would likely have a different result today based on new discovered vulnerabilities.  Also, Red Hat occasionally updates their ubi images, so if we wait long enough, they'll patch the libexpat issues in a month or two.

I saw 4 total items, which I posted in the issue [758](https://github.ibm.com/ai-foundation/watson-fm-stack-tracker/issues/758)
Two were expat, one was a django update that I could not easily find a fix for, and one was a private key that I remove using the dockerfile.

When we were going over the detailed list of items with Red Hat and the product team early a few weeks ago, one of the items was updating the OS as part of the build.  Let me see if I can find that doc and we can decide if we want to address it or not...

Reply:
If you can find that doc, that would be great and can approve from there. Thanks Jim!

Reply:
I don't think we hardcode the requirements above and a new image rebuild would probably pick up newer versions of the above as needed. Although I thought we are using python 3.11 and the above shows a vulnerability in python 3.9?

Reply:
Yes, I looked into it, and python 3.9 is installed as a requirement for `dnf` which is used to install multiple components.  I'll see if I can address in a future PR.